app/javascript/dashboard/components/widgets/AIAssistanceButton.vue

```diff
@@ -1,48 +1,85 @@
 <template>
-  <div v-if="isAIIntegrationEnabled" class="position-relative">
-    <woot-button
-      v-tooltip.top-end="$t('INTEGRATION_SETTINGS.OPEN_AI.AI_ASSIST')"
-      icon="wand"
-      color-scheme="secondary"
-      variant="smooth"
-      size="small"
-      @click="openAIAssist"
-    />
-    <woot-modal
-      :show.sync="showAIAssistanceModal"
-      :on-close="hideAIAssistanceModal"
-    >
-      <AIAssistanceModal
-        :ai-option="aiOption"
-        @apply-text="insertText"
-        @close="hideAIAssistanceModal"
+  <div v-if="!isFetchingAppIntegrations">
+    <div v-if="isAIIntegrationEnabled" class="position-relative">
+      <AIAssistanceCTAButton
+        v-if="shouldShowAIAssistCTAButton"
+        @click="openAIAssist"
+      />
+      <woot-button
+        v-else
+        v-tooltip.top-end="$t('INTEGRATION_SETTINGS.OPEN_AI.AI_ASSIST')"
+        icon="wand"
+        color-scheme="secondary"
+        variant="smooth"
+        size="small"
+        @click="openAIAssist"
       />
-    </woot-modal>
+      <woot-modal
```

Comment:
can we improve the condition check `<div v-if="!isFetchingAppIntegrations">` to consider error case as well, if not should we put the model, outside of the condition for `<div v-if="!isFetchingAppIntegrations">`

Reply:
I added this condition to hide the CTA button until the integrations are fetched.